### PR TITLE
Fix/gradle cleanup

### PR DESCRIPTION
**Pull request by ritatferreira**
- Gradle files cleanup
- New nullIfEmpty extension

### CODE REVIEW

File: rxjava/src/main/kotlin/com/mindera/skeletoid/rxjava/Observable.kt [hunk 1]
```diff
@@ -39,7 +39,12 @@ fun <T : Any> Observable<T>.createUniqueConcurrentRequestCache(
         return requestMap[key] as Observable<T>
     }
 
-    val obs = this.doFinally {
+    val obs = this.doAfterNext{
+        requestMap.remove(key)
+    }
```
Comment (bruno-garces) [inline rxjava/src/main/kotlin/com/mindera/skeletoid/rxjava/Observable.kt:44]: Why this change?

Comment (neteinstein) [reply rxjava/src/main/kotlin/com/mindera/skeletoid/rxjava/Observable.kt:44]: This was added to prevent concurrency issues, if just done at finally, if you are using heavy concurrency it would be removed after it should.

/cc @Jmvcaseiro

**Review by bruno-garces — APPROVED** [review]

**Review by neteinstein — APPROVED** [review]

**Review by rcosteira79 — APPROVED** [review]

**Review by MajaTrzebiatowska — APPROVED** [review]
